components/draw/draw.service.ts

```diff
@@ -335,12 +348,47 @@ export class HsDrawService {
     }
     this.drawableLayers = drawables;
   }
+  /**
+   * @function removeLayer
+   * @memberOf HsDrawController
+   * @description Removes selected drawing layer from both Layermanager and Layman
+   */
+  async removeLayer() {
+    const dialog = this.HsDialogContainerService.create(
+      HsConfirmDialogComponent,
+      {
+        message: this.TranslateService.instant('Really delete this layer?'),
+        title: this.TranslateService.instant('Confirm delete'),
+      }
+    );
+    const confirmed = await dialog.waitResult();
+    if (confirmed == 'yes') {
+      this.HsMapService.map.removeLayer(this.selectedLayer);
+      if (this.selectedLayer.get('synchronize') == true) {
```

Comment:
Would be better to move insides of this if condition to layman service to seperate concerns. Drawing service shouldnt manage layers and do synchronization to endpoints.

Reply:
15564b091777044edb8d270dce08ca85724d0e90